File: IntelligenceTraffic_Scala_v1.0/src/main/scala/com/uek/bigdata/DAO/inter/IMonitorDao.scala
package com.uek.bigdata.DAO.inter

import com.uek.bigdata.daomain.{MonitorState, TopNMonitor2CarCount, TopNMonitorDetailInfo}

import scala.collection.mutable.ListBuffer

/**
  * 类名称: IMonitorStateDao
  * 类描述:
  *
  * @Time 2018/12/1 22:11
  * @author 武建谦
  */
trait IMonitorDao {
  /**
    * 将自定义累加器记录的五种状态插入到数据库：
    *
    * @param monitorState
    */
  def insertMonitorState(monitorState: MonitorState): Unit

  /**
    * 获取车流量topN的卡口
    * @param topns
    */
  def insertTopNMonitor2CarCount(topns:ListBuffer[TopNMonitor2CarCount]) : Unit

  /**
    * 将topN卡口中的详细信息插入到数据库中
    * @param topNMonitorDetailInfos
    */
  def insertTopNMonitorDetailInfo(topNMonitorDetailInfos:ListBuffer[TopNMonitorDetailInfo]) : Unit

}

File: IntelligenceTraffic_v2.0/src/main/java/com/uek/bigdata/DAO/inter/IMonitorDAO.java
package com.uek.bigdata.DAO.inter;

import com.uek.bigdata.daomain.MonitorState;
import com.uek.bigdata.daomain.TopNMonitor2CarCount;
import com.uek.bigdata.daomain.TopNMonitorDetailInfo;

import java.util.List;

/**
 * 类名称: IMonitorDAO
 * 类描述:
 *
 * @author 武建谦
 * @Time 2018/11/26 16:43
 */
public interface IMonitorDAO {
    /**
     * 将自定义累加器记录的五种状态插入到数据库：
     * @param monitorState
     */
    void insertMonitorState(MonitorState monitorState);

    /**
     * 将车流量最大的topN卡口信息插入到数据库：
     * @param topNMonitor2CarCounts
     */
    void insertBatchTopN(List<TopNMonitor2CarCount> topNMonitor2CarCounts);

    /**
     *将车流量最大的topN卡口的详细信息插入到数据库：
     */
    void insertBatchMonitorDetails(List<TopNMonitorDetailInfo> topNMonitorDetailInfo);
    /**
     * 被高速通过的前5个卡口，每个卡口下的最快的10辆车的信息插入到数据库：
     * @param topNMonitorDetailInfos
     */
    void insertBatchTop10Details(List<TopNMonitorDetailInfo> topNMonitorDetailInfos);
}
